PREPROCESSING DATA STUDENTS DENGAN MAYORITAS ATRIBUT NUMERIK
- cleaning data
- analisis data
- mengatasi mising value
- mengatasi outlier
- mengatasi duplikasi

A. Import dataset

In [51]:
import os
import pandas as pd

data_folder = "/Users/althaaframadhan/Documents/Skripsi/dataset/Students Social Media Addiction.csv"
csv_files = [os.path.basename(data_folder)]
dataframes = {}

# Read the CSV file into a dataframe
df_name = os.path.splitext(csv_files[0])[0]
dataframes[df_name] = pd.read_csv(data_folder)


B. Melihat 10 baris awal dataset untuk dilakukan analisis bentuk dataset

In [52]:
df = dataframes['Students Social Media Addiction']
print(df.head(10))

   Age  Gender Academic_Level      Country  Avg_Daily_Usage_Hours  \
0   19  Female  Undergraduate   Bangladesh                    5.2   
1   22    Male       Graduate        India                    2.1   
2   20  Female  Undergraduate          USA                    6.0   
3   18    Male    High School           UK                    3.0   
4   21    Male       Graduate       Canada                    4.5   
5   19  Female  Undergraduate    Australia                    7.2   
6   23    Male       Graduate      Germany                    1.5   
7   20  Female  Undergraduate       Brazil                    5.8   
8   18    Male    High School        Japan                    4.0   
9   21  Female       Graduate  South Korea                    3.3   

  Most_Used_Platform Affects_Academic_Performance  Sleep_Hours_Per_Night  \
0          Instagram                          Yes                    6.5   
1            Twitter                           No                    7.5   
2           

C. Melihat atribut dan bentuk nilai setiap atribut

In [53]:
df.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 705 entries, 0 to 704
Data columns (total 11 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   Age                           705 non-null    int64  
 1   Gender                        705 non-null    object 
 2   Academic_Level                705 non-null    object 
 3   Country                       705 non-null    object 
 4   Avg_Daily_Usage_Hours         705 non-null    float64
 5   Most_Used_Platform            705 non-null    object 
 6   Affects_Academic_Performance  705 non-null    object 
 7   Sleep_Hours_Per_Night         705 non-null    float64
 8   Mental_Health_Score           705 non-null    int64  
 9   Conflicts_Over_Social_Media   705 non-null    int64  
 10  Addicted_Score                705 non-null    int64  
dtypes: float64(2), int64(4), object(5)
memory usage: 60.7+ KB


In [54]:
print("duplikasi :", df.duplicated().sum())

duplikasi : 1


In [55]:
duplicates = df[df.duplicated(keep=False)]
print(duplicates)


     Age Gender Academic_Level Country  Avg_Daily_Usage_Hours  \
241   21   Male       Graduate   Italy                    5.4   
273   21   Male       Graduate   Italy                    5.4   

    Most_Used_Platform Affects_Academic_Performance  Sleep_Hours_Per_Night  \
241          Instagram                          Yes                    6.7   
273          Instagram                          Yes                    6.7   

     Mental_Health_Score  Conflicts_Over_Social_Media  Addicted_Score  
241                    5                            3               7  
273                    5                            3               7  


pada df.info() memberikan output nama atribut, dan setiap atribut memiliki 705 nilai data dan tidak ada missing values dan duplikasi data

D. subset data menjadi 200 data yang akan digunakan pada pengembangan model

In [56]:
df = dataframes['Students Social Media Addiction']
students_raw = df.copy()
students_subset = students_raw.sample(n=200, random_state=42).reset_index(drop=True)
students_subset.to_csv("/Users/althaaframadhan/Documents/Skripsi/dataset/students_200.csv", index=False)

print("Subset created and saved successfully.")

Subset created and saved successfully.


E. menampilkan 10 data awal dari 200 data yang telah diambil secara acak dari dataset awal

In [57]:
students_subset = pd.read_csv("/Users/althaaframadhan/Documents/Skripsi/dataset/students_200.csv")
print(students_subset.head(10))


   Age  Gender Academic_Level      Country  Avg_Daily_Usage_Hours  \
0   20  Female  Undergraduate       France                    3.0   
1   19  Female  Undergraduate   Azerbaijan                    4.8   
2   22  Female       Graduate   Kyrgyzstan                    2.9   
3   19  Female  Undergraduate   Bangladesh                    4.2   
4   22    Male       Graduate      Germany                    3.1   
5   20  Female  Undergraduate          USA                    6.8   
6   19  Female  Undergraduate        Japan                    3.6   
7   19  Female  Undergraduate  Switzerland                    4.0   
8   23  Female       Graduate      Finland                    4.4   
9   20    Male  Undergraduate    Sri Lanka                    5.9   

  Most_Used_Platform Affects_Academic_Performance  Sleep_Hours_Per_Night  \
0          Instagram                           No                    9.0   
1             TikTok                          Yes                    5.7   
2           

In [58]:
df = df.drop_duplicates()
df.duplicated().sum()

0

In [59]:
print(students_subset.info())
print("duplikasi :", df.duplicated().sum())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 200 entries, 0 to 199
Data columns (total 11 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   Age                           200 non-null    int64  
 1   Gender                        200 non-null    object 
 2   Academic_Level                200 non-null    object 
 3   Country                       200 non-null    object 
 4   Avg_Daily_Usage_Hours         200 non-null    float64
 5   Most_Used_Platform            200 non-null    object 
 6   Affects_Academic_Performance  200 non-null    object 
 7   Sleep_Hours_Per_Night         200 non-null    float64
 8   Mental_Health_Score           200 non-null    int64  
 9   Conflicts_Over_Social_Media   200 non-null    int64  
 10  Addicted_Score                200 non-null    int64  
dtypes: float64(2), int64(4), object(5)
memory usage: 17.3+ KB
None
duplikasi : 0


F. cek outliers pada data

In [60]:
import numpy as np

def detect_outliers_iqr(df, column):
    Q1 = df[column].quantile(0.25)
    Q3 = df[column].quantile(0.75)
    IQR = Q3 - Q1
    lower, upper = Q1 - 1.5 * IQR, Q3 + 1.5 * IQR
    outliers = df[(df[column] < lower) | (df[column] > upper)]
    return outliers

numeric_students = students_subset.select_dtypes(include=np.number).columns


for col in numeric_students:
    out = detect_outliers_iqr(students_subset, col)
    print(f"{col}: {len(out)} outliers")




Age: 0 outliers
Avg_Daily_Usage_Hours: 1 outliers
Sleep_Hours_Per_Night: 0 outliers
Mental_Health_Score: 0 outliers
Conflicts_Over_Social_Media: 0 outliers
Addicted_Score: 0 outliers


pada output tersebut mengetahui adanya 1 outliers pada atribut Avg_Daily_Usage_Hours

G. cek nilai outliers pada atribut Avg_Daily_Usage_Hours

In [61]:
# Get the outliers for Avg_Daily_Usage_Hours
outliers = detect_outliers_iqr(students_subset, 'Avg_Daily_Usage_Hours')

# Display the outliers with relevant columns
print("Outliers in Avg_Daily_Usage_Hours:")
print(outliers[['Avg_Daily_Usage_Hours', 'Age', 'Gender', 'Academic_Level', 'Country']])

# Calculate the IQR values for reference
Q1 = students_subset['Avg_Daily_Usage_Hours'].quantile(0.25)
Q3 = students_subset['Avg_Daily_Usage_Hours'].quantile(0.75)
IQR = Q3 - Q1

print(f"\nSummary Statistics:")
print(f"Q1: {Q1:.2f}")
print(f"Q3: {Q3:.2f}")
print(f"IQR: {IQR:.2f}")
print(f"Lower bound: {(Q1 - 1.5 * IQR):.2f}")
print(f"Upper bound: {(Q3 + 1.5 * IQR):.2f}")

Outliers in Avg_Daily_Usage_Hours:
     Avg_Daily_Usage_Hours  Age Gender Academic_Level  Country
128                    1.5   23   Male       Graduate  Germany

Summary Statistics:
Q1: 4.20
Q3: 5.90
IQR: 1.70
Lower bound: 1.65
Upper bound: 8.45


nilai yang ditampilkan pada data 128 memiliki nilai yang rendah dari rata rata statistik yang telah ditentukan, tetapi nilai data tidak terlalu jauh dengan rata rata statistik sehingga data dibiarkan untuk melatih model dengan adanya 1 outliers